### DISCUSSION TITLE
mix SSR and CSR?

### DISCUSSION THREAD

**Discussion by fimbault** [frandiox/vitedge#21] [Q&A] [answered]
In existing framework such as nextjs, it's not easy to create complex applications where a part of the website is static, and a part is dynamic. See for instance https://github.com/vercel/next.js/discussions/19606

Seeing https://vitedge.netlify.app/props.html#alternatives-to-page-props, I'm wondering if vitedge would allow that?

ps: not looking for a real isomorphic framework

**Comment by frandiox** [comment, marked as answer]
@fimbault Hello! In Vitedge there are no "build time props". However, since it caches at the edge, as long as you provide a long enough max-age for the props, it can be considered as static. And then yes, you can use API calls to get dynamic data without caching if you want that. Caching the rendered HTML can also be controlled separately but I guess you've noticed that already.

I recommend testing it quickly using https://github.com/subhendukundu/vitedge-react-template and see if it fits your needs.
